module/VuFind/src/VuFind/View/Helper/Root/Record.php

```diff
@@ -297,6 +297,37 @@ public function getPreviewIds()
         return $idClasses;
     }
 
+    /**
+     * Formats the title. This method will return both the regular title and an
+     * alternate title if both are present, otherwise it will return either the
+     * regular title or the alternate title if only one is present. This method
+     * tries to return the highlighted title first but will return the unhighlighted
+     * title if a one is not present. Returns 'Title not available' if there is no
+     * title.
+     *
+     * @param string $title               Title not highlighted
+     * @param string $altTitle            Alternate title not highlighted
+     * @param string $highlightedTitle    Highlighted title
+     * @param string $highlightedAltTitle Alternate highlighted title
+     *
+     * @return string
+     */
+    protected function formatTitle(
+        $title, $altTitle, $highlightedTitle, $highlightedAltTitle
+    ) {
+        $format = '%s<br/>%s';
+        $title = !empty($highlightedTitle) ? $highlightedTitle : $title;
+        $altTitle = !empty($highlightedAltTitle) ? $highlightedAltTitle : $altTitle;
+        if ($title && $altTitle) {
+            return sprintf($format, $title, $altTitle);
+        }
+        if ($title || $altTitle) {
+            return $title ?? $altTitle;
+        }
+        $transEsc = $this->getView()->plugin('transEsc');
+        return $transEsc('Title not available');
+    }
```

Comment:
I wonder if this function can't be generalized such that it might work for all linked field pairs. I think I'll get a feel for that as I add other single value fields.

Reply:
Agreed, seems possible.